x11-misc/sddm/sddm-0.19.0_p20201230.ebuild

```diff
@@ -0,0 +1,172 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+COMMIT=5825c89444626755ca7ce5eef454c4888131a12b
+inherit cmake linux-info pam systemd tmpfiles
+
+DESCRIPTION="Simple Desktop Display Manager"
+HOMEPAGE="https://github.com/sddm/sddm"
+SRC_URI="https://github.com/${PN}/${PN}/archive/${COMMIT}.tar.gz -> ${P}.tar.gz"
+S="${WORKDIR}/${PN}-${COMMIT}"
+
+LICENSE="GPL-2+ MIT CC-BY-3.0 CC-BY-SA-3.0 public-domain"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc64 ~x86"
+IUSE="+elogind kernel-randomness openrc-init systemd test"
+
+REQUIRED_USE="^^ ( elogind systemd )"
+RESTRICT="!test? ( test )"
+
+BDEPEND="
+	dev-python/docutils
+	dev-qt/linguist-tools:5
+	kde-frameworks/extra-cmake-modules:5
+	virtual/pkgconfig
+"
+COMMON_DEPEND="
+	dev-qt/qtcore:5
+	dev-qt/qtdbus:5
+	dev-qt/qtdeclarative:5
+	dev-qt/qtgui:5
+	dev-qt/qtnetwork:5
+	sys-libs/pam
+	x11-base/xorg-server
+	x11-libs/libxcb[xkb]
+	elogind? ( sys-auth/elogind )
+	systemd? ( sys-apps/systemd:= )
+	!systemd? ( sys-power/upower )
+"
+DEPEND="${COMMON_DEPEND}
+	test? ( dev-qt/qttest:5 )
+"
+RDEPEND="${COMMON_DEPEND}
+	acct-group/sddm
+	acct-user/sddm
+	!kernel-randomness? ( || (
```

Comment:
I don't like this "kernel-randomness" USE flag. Pulling in entropy daemons as optional runtime deps is a violation of QA policy, and really just kind of weird. Please drop this.

Reply:
While I agree with that in principle, there's a real issue tried to be solved here, where without one of those daemons sddm will hang for some time (minus e.g. random keyboard presses) but the better part of ~50% systems will work fine. The only alternative then is to hard-depend on one-of entropy daemons.

Reply:
If you want to advertise to users that they need good entropy on boot, write a news item or something. Halfway-enforcing this via a dependency in a random X11 package is the wrong way to go about this.

Reply:
A news item is insufficient. It has been a longstanding problem, and it will probably not go away soon, so unless we make that QA exception here we are down to `ewarn`/`elog` that no one reads (but at least we can say "told you so").

Reply:
Depending on entropy daemons that few people will configure to start at boot does nothing to "solve" the issue either.

Also, the lack of an entropy daemon does not imply that the kernel will be entropy starved at boot. For example, I'm running an older AMD processor that does not support the RDRAND instructions, and the kernel is still able to gather sufficient entropy within a few seconds on boot (without an entropy daemon).

A news item on the subject in general would be much more effective I think.

Reply:
> Depending on entropy daemons that few people will configure to start at boot does nothing to "solve" the issue either.

That I can agree on, which is why it is covered by `pkg_postinst` message.